### PR TITLE
[frontend] use doc_code for connector errors (#8614)

### LINKED ISSUE #8614: Connectors/Ingestors and GUI error logs messages improvements
Improve our error log format to include more detail about a connector ingestion error.
Improve the UI to better display error log messages.

### PR DESCRIPTION
**Pull request by frapuks**
### Proposed changes

* Add `doc_code` in some connector errors
* Use this doc_code in the data/ingestion/connector / (select one)  / (open error drawer) / errors table
* Fallback on error name

### Related issues

* close #8614 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/data/connectors/ConnectorWorksErrorLine.tsx [hunk 1]
```diff
@@ -58,19 +58,19 @@ const ConnectorWorksErrorLine: FunctionComponent<ConnectorWorksErrorLineProps> =
       <TableRow key={error.rawError.timestamp}>
         <TableCell>{nsdt(error.rawError.timestamp)}</TableCell>
         <TableCell>
-          {error.isParsed ? (
-            <a href={`https://docs.opencti.io/latest/deployment/troubleshooting/#${error.parsedError.category}`} target="_blank" rel="noreferrer">{error.parsedError.category}</a>
+          {error.isParsed && error.parsedError.doc_code ? (
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/data/connectors/ConnectorWorksErrorLine.tsx:61]: how will it work for previous errors that don't have a doc_code yet ? will it display only unknown ? the documentation will change and won't be retrocompatible I guess ?

File: opencti-platform/opencti-front/src/private/components/data/connectors/parseWorkErrors.tsx [hunk 2]
```diff
@@ -103,76 +104,115 @@ export interface PartialParsedWorkMessage {
...
+    const parsedSource = JSON5.parse(error.source ?? '');
+    const source = parsedSource.type === 'bundle' ? parsedSource.objects[0] : parsedSource;
+    const message = JSON5.parse((error.message ?? ''));
+    const entityId = source.id;
+    const fromId = source.source_ref;
+    const toId = source.target_ref;
+
+    const parsedError = {
+      category: message.name,
+      doc_code: message.doc_code ?? message.name,
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/data/connectors/parseWorkErrors.tsx:151]: Oh I see here there is a fallback on the name. Forget about my comment, I'll resolve it now :)

File: opencti-platform/opencti-front/src/private/components/data/connectors/parseWorkErrors.test.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      level: 'Unclassified',
+      rawError: {
+        timestamp: null,
+        message: null,
+        sequence: null,
+        source: null,
+      },
+    });
+  });
+});
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/data/connectors/parseWorkErrors.test.ts:149]: Thanks for the tests !!!! 😍

**Review by marieflorescontact — APPROVED** [review]
locally tested ok 👌

**Pull request merged**